### PR TITLE
Support publishing events to an app's log stream

### PR DESCRIPTION
**Pull request by mhahn**
Addresses: https://github.com/remind101/empire/issues/832

I wasn't sure about the name for the event streamer.

Calling it `kinesis` seemed off, because I would assume that a true `kinesis` event backend would take a stream name as a config and write everything to that stream.

### CODE REVIEW

File: events/app/app.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+// EventStream is an implementation of the empire.EventStream interface backed
+// by Kinesis.
+type EventStream struct {
+	empire  *empire.Empire
+	kinesis kinesisClient
+}
+
+type Event interface {
```
Comment (ejholmes) [inline events/app/app.go]: Minor, but might be nice to move this interface below https://github.com/remind101/empire/blob/master/events.go#L191-L198 as:

```go
// An AppEvent is an Event that relates to a specific app.
type AppEvent interface {
        Event
        App() *App // AppID is fine too.
}
```

File: events.go [hunk 2]
```diff
@@ -215,22 +251,24 @@ var NullEventStream = EventStreamFunc(func(event Event) error {
...
+	events  chan Event
 }
 
 // AsyncEvents returns a new AsyncEventStream that will buffer upto 100 events
 // before applying backpressure.
-func AsyncEvents(e EventStream) *AsyncEventStream {
-	s := &AsyncEventStream{
-		EventStream: e,
-		events:      make(chan Event, 100),
+func AsyncEvents(s []EventStream) *AsyncEventStream {
```
Comment (ejholmes) [inline events.go]: You could keep this as an `EventStream` and just make `[]EventStream` implement the EventStream interface:

```go
// MutliEventStream is an EventStream implementation that publishes the event to multiple EventStreams, returning any errors after publishing to all streams.
type MultiEventStream []EventStream

func (streams MultiEventStream) PublishEvent(event Event) error {
	var errors []error
	for _, s := range streams {
		errors = append(errors, s.PublishEvent(event))
	}

	if len(errors) > 0 {
		// Probably a better way to generate a MultiError here.
		return fmt.Errorf("errors publishing to event streams: %v", errors)
	}

	return nil
}
```

**Comment by ejholmes** [comment]
This is awesome. Other than those comments, 👍.

File: events.go (hunk 2 above)
Comment (mhahn) [inline events.go]: woah, thats cool!

File: events/app/app.go (hunk 1 above)
Comment (mhahn) [inline events/app/app.go]: yea i like that

File: events.go (hunk 2 above)
Comment (mhahn) [inline events.go]: https://github.com/hashicorp/go-multierror seems to handle multierrors cleanly so i went with that instead of reimplementing something.

**Comment by ejholmes** [comment]
Awesome 👍

**Pull request merged**